From e86da1887fec3df7f5003ea52fd93d4d63d255d3 Mon Sep 17 00:00:00 2001
From: Brian C. Lane <bcl@redhat.com>
Date: Fri, 17 Jun 2011 17:48:16 -0700
Subject: [PATCH 1/4] Fix snap radius so that it is using half (#665496)

The snap radius didn't match the documentation, it has been using +/-
unit size instead +/- 0.5 * unit (eg. 500KB for a MB unit). This caused
problems when specifying 1MB, 1GB, etc. as a partition start or end
resulting in partitions being created that were nowhere near the
specified size.

* libparted/unit.c (ped_unit_parse_custom): divide radius by 2
---
 libparted/unit.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/libparted/unit.c b/libparted/unit.c
index b9a252a..0526688 100644
--- a/libparted/unit.c
+++ b/libparted/unit.c
@@ -527,7 +527,7 @@ ped_unit_parse_custom (const char* str, const PedDevice* dev, PedUnit unit,
 	}
 
 	unit_size = ped_unit_get_size (dev, unit);
-	radius = ped_div_round_up (unit_size, dev->sector_size) - 1;
+	radius = (ped_div_round_up (unit_size, dev->sector_size)  / 2) - 1;
 	if (radius < 0)
 		radius = 0;
 
-- 
1.7.4.4

